src/app/digital/models/ioobjects/inputs/ConstantNumber.ts

```diff
@@ -0,0 +1,59 @@
+import {DigitalComponent} from "digital/models/DigitalComponent";
+import {ClampedValue} from "math/ClampedValue";
+import {DecimalToBCD} from "math/MathUtils";
+import {serializable, serialize} from "serialeazy";
+import {V} from "Vector";
+
+/**
+ * A constant number input. This allows the user to control a 4 bit constant
+ * output by using a single hexadecimal digit as input.
+ */
+@serializable("ConstantNumber")
+export class ConstantNumber extends DigitalComponent {
+
+    // The number used to determine the output values
+    @serialize
+    private inputNumber : number;
```

Comment:
nit: Remove space before `:`

nit: Let's call this just `inputNum` to stick to similarly named values in the codebase